lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -77,15 +82,45 @@ def fetch_bindnetloc
     Rex::Socket.to_authority(fetch_bindhost, fetch_bindport)
   end
 
+
   def generate(opts = {})
     opts[:arch] ||= module_info['AdaptedArch']
     opts[:code] = super
     @srvexe = generate_payload_exe(opts)
-    cmd = generate_fetch_commands
+    if datastore['FETCH_PIPE']
+      supported_binaries = %w[CURL]
+      supported_binaries = %w[WGET CURL] if linux?
```

Comment:
```suggestion
      supported_binaries << "WGET" if linux?
```

Something like this makes things a bit more obvious I think.

Reply:
Then maybe a construct like a switch-case would be better, like

```
switch
  case windows:
   …
  case linux:
   …
```

Reply:
This makes a lot of sense.  A little overkill now, but in the back of my mind, I want to add windows_psh payloads which are going to need a third set of requirements.